pkg/hub/deployer/helm/helm.go

```diff
@@ -273,32 +275,68 @@ func (deployer *Deployer) populateHelmRelease(desc *appsapi.Description) error {
 		if err != nil {
 			return err
 		}
+		defaultLabels := map[string]string{
+			known.ObjectCreatedByLabel: known.ClusternetHubName,
+			known.ConfigKindLabel:      descriptionKind.Kind,
+			known.ConfigNameLabel:      desc.Name,
+			known.ConfigNamespaceLabel: desc.Namespace,
+			known.ConfigUIDLabel:       string(desc.UID),
+			known.ClusterIDLabel:       desc.Labels[known.ClusterIDLabel],
+			known.ClusterNameLabel:     desc.Labels[known.ClusterNameLabel],
+			// add subscription info
+			known.ConfigSubscriptionNameLabel:      desc.Labels[known.ConfigSubscriptionNameLabel],
+			known.ConfigSubscriptionNamespaceLabel: desc.Labels[known.ConfigSubscriptionNamespaceLabel],
+			known.ConfigSubscriptionUIDLabel:       desc.Labels[known.ConfigSubscriptionUIDLabel],
+		}
+		copyChart := chart.DeepCopy()
+		if copyChart.Labels == nil {
+			copyChart.Labels = make(map[string]string)
+		}
+		// add default labels to copyChart
+		for key, value := range defaultLabels {
+			copyChart.Labels[key] = value
+		}
+		// backward compatibility for existing old description cr resource
+		if desc.Spec.ChartRaw != nil && len(desc.Spec.ChartRaw) == len(desc.Spec.Charts) &&
```

Comment:
Please simplify this piece of codes.

Reply:
IMO, we can marshal the original chart spec and apply the override, then unmarshal it.

Reply:
I need to filter "spec.Chart" and "spec.TargetNamespac", so can't directly override

Reply:
> I need to filter "spec.Chart" and "spec.TargetNamespac", so can't directly override

We do have other way to handle this. Such as, you can set the original value back after applying all the overrides.